internal/envtest/webhooks.go

```diff
@@ -97,14 +99,38 @@ func appendWebhookConfiguration(mutatingWebhooks []client.Object, validatingWebh
 			// update the name in metadata
 			if o.GetName() == mutatingwebhook {
 				o.SetName(strings.Join([]string{mutatingwebhook, "-", tag}, ""))
-				mutatingWebhooks = append(mutatingWebhooks, &o)
+
+				webhook := &admissionv1.MutatingWebhookConfiguration{}
+				if err := scheme.Scheme.Convert(&o, webhook, nil); err != nil {
+					klog.Fatalf("failed to convert MutatingWebhookConfiguration %s", o.GetName())
+				}
+
+				gvk, err := apiutil.GVKForObject(webhook, scheme.Scheme)
```

Comment:
Note: I think we should fix the controller-runtime code so we don't have to do this here.

Reply:
Fix here: https://github.com/kubernetes-sigs/controller-runtime/pull/1665

Reply:
Update: I think we have to wait for https://github.com/kubernetes-sigs/controller-runtime/pull/1666 anyway

Reply:
PRs merged, I dropped the redundant code